types/timestamp_test.go

```diff
@@ -135,3 +135,37 @@ func TestTimestampString(t *testing.T) {
 func utcDate(year, month, day int) time.Time {
 	return time.Date(year, time.Month(month), day, 0, 0, 0, 0, time.UTC)
 }
+
+func BenchmarkStdTimeUnmarshal(b *testing.B) {
+	data, err := StdTimeMarshal(time.Time{})
+	if err != nil {
+		panic(err)
+	}
+
+	b.ReportAllocs()
+	b.RunParallel(func(pb *testing.PB) {
+		var t time.Time
+		for pb.Next() {
+			if err := StdTimeUnmarshal(&t, data); err != nil {
+				panic(err)
+			}
+		}
+	})
+}
+
+func TestStdTimeUnmarshalAllocs(t *testing.T) {
+	data, err := StdTimeMarshal(time.Time{})
```

Comment:
I'll add another AllocsPer with a time.Now() to prove allocs don't happen for both the zero value and non-zero.